src/Npgsql/TypeMapping/ConnectorTypeMapper.cs

```diff
@@ -415,7 +415,7 @@ Type GetUnderlyingType(Type t)
     internal bool TryGetMapping(PostgresType pgType, [NotNullWhen(true)] out TypeMappingInfo? mapping)
     {
         foreach (var resolver in _resolvers)
-            if ((mapping = resolver.GetMappingByDataTypeName(pgType.Name)) is not null)
+            if ((mapping = resolver.GetMappingByDataTypeName(pgType.DisplayName)) is not null)
```

Comment:
While that's definitely more correct to pass the full name, it's technically a breaking change. On the other hand, it's only used with `Schema` api, which isn't that popular, so it might be fine. @roji what's your opinion?